### DISCUSSION TITLE
Error keeping leadership

### DISCUSSION THREAD

**Discussion by ciokan** [riverqueue/river#510] [Q&A] [answered]
I get this after a short amount of time. I am connected to Supabase and my client uses `PollOnly`. The connection is established but fails after a minute or so and no new job can perform.

`Elector: Error keeping leadership" client_id=rms-Mac-Pro err="context deadline exceeded"`. Connection is done on port `5432` which is Session mode. On transaction mode I get different errors related to stored procedures or something and my qorker does not even come up.

Any ideas?

**Comment by bgentry** [comment, marked as answer]
> The connection is established but fails after a minute or so and no new job can perform.

This bit is interesting. [Leadership](https://riverqueue.com/docs/leader-election) is about which node gets to perform various [maintenance tasks](https://riverqueue.com/docs/maintenance-services) and in no way affects a client's ability to work jobs. If you're seeing issues with job execution the issue is likely something else, such as you closing the context provided to your `Client`, or issues with your database pool maintaining healthy connections.

**Reply by ciokan** [reply]
good catch, I was using an inherited context of which I thought it is long lived. Many thanks.
